python/fusion_engine_client/messages/control.py

```diff
@@ -223,24 +229,67 @@ class ResetRequest(MessagePayload):
     # - All runtime data (GNSS corrections (@ref RESET_GNSS_CORRECTIONS), etc.)
     # - Position, velocity, orientation (@ref RESET_POSITION_DATA)
     # - Fast IMU corrections (@ref RESET_FAST_IMU_CORRECTIONS)
-    # - GNSS measurement engine (@ref RESTART_GNSS_MEASUREMENT_ENGINE)
+    # - GNSS ephemeris data (@ref RESET_EPHEMERIS)
     #
     # Not reset:
     # - Training parameters (slowly estimated IMU corrections, temperature
     #   compensation, etc.; @ref RESET_NAVIGATION_ENGINE_DATA)
     # - Calibration data (@ref RESET_CALIBRATION_DATA)
     # - User configuration settings (@ref RESET_CONFIG)
+    # - Reboot GNSS measurement engine (@ref RESTART_GNSS_MEASUREMENT_ENGINE)
     # - Reboot navigation processor (@ref REBOOT_NAVIGATION_PROCESSOR)
     #
     # @note
     # To reset training or calibration data as well, set the @ref
     # RESET_NAVIGATION_ENGINE_DATA and @ref RESET_CALIBRATION_DATA bits.
-    COLD_START = 0x01000FFF
+    COLD_START = 0x00000FFF
 
     ##
     # Restart mask to set all persistent data, including calibration and user configuration, back to factory defaults.
     FACTORY_RESET = 0xFFFFFFFF
 
+    # Perform a @ref HOT_START, and also flush internal data buffers to guarantee
+    # deterministic performance for post-processing and support.
+    #
+    # To be reset:
+    # - The navigation engine (@ref RESTART_NAVIGATION_ENGINE)
+    # - All runtime data (GNSS corrections (@ref RESET_GNSS_CORRECTIONS), etc.)
+    # - GNSS ephemeris data (@ref RESET_EPHEMERIS)
+    #
+    # Not reset:
+    # - Position, velocity, orientation (@ref RESET_POSITION_DATA)
+    # - Fast IMU corrections (@ref RESET_FAST_IMU_CORRECTIONS)
+    # - Training parameters (slowly estimated IMU corrections, temperature
+    #   compensation, etc.; @ref RESET_NAVIGATION_ENGINE_DATA)
+    # - Calibration data (@ref RESET_CALIBRATION_DATA)
+    # - User configuration settings (@ref RESET_CONFIG)
+    # - Reboot GNSS measurement engine (@ref RESTART_GNSS_MEASUREMENT_ENGINE)
+    # - Reboot navigation processor (@ref REBOOT_NAVIGATION_PROCESSOR)
+    DETERMINISTIC_RESET = 0x00002FF
```

Comment:
perhaps we call this DIAGNOSTIC_LOG_RESET?

Reply:
(your name isnt bad, im just thinking if its more functional or if its more the "what this actually does".

Reply:
Yeah that seems good